.azure-pipelines/ultimate-pipeline.yml

```diff
@@ -3094,6 +3094,21 @@ stages:
           displayName: Publish release artifacts
           artifact: $(tracer_version)-release-artifacts
 
+        # lib-injection images
```

Comment:
In the end-to-end flow, the GitHub container registry WILL e used in the production path.

First, the regular CI on merge to `master` will produce these images in the GitHub Container Registry, then our GitLab pipeline will pull these images and re-push them to the official Datadog docker.io/gcr.io registries. Here's an example of how Java does it: https://github.com/DataDog/dd-trace-java/blob/e34349483151e248599106cad27266c0c3b5d870/.gitlab-ci.yml#L160-L176

Reply:
So in the dd-trace-java `.gitlab-ci.yml` that I linked, they only publish an official build image when the job is triggered by a tag with the format `v.*` (and it does this by looking up the image via commit SHA on the GitHub Container Registry and publishing that to the official registries). We can do the same thing, but in this PR I haven't included the GitLab CI changes yet so this PR won't publish anything to official channels yet.

The other question is, on what frequency do we want to publish to the GitHub Container Registry? I'm guessing only on merges to master, right? If so, I have a few more changes I need to make before merging.

Reply:
Thanks a lot for looking into this

> what frequency do we want to publish to the GitHub Container Registry? I'm guessing only on merges to master, right?

Yes. 

> the job is triggered by a tag with the format v.*

If the gitlab job is triggered with the tag, it means the publish will happen automatically when we release as we create the tag with the release. It means the upload will happen before the release is published, which is what happen for Nuget as well, so I'm fine with it. 
We should also make sure that the alerting we have in Datadog Ci App catches all gitlab build (I think it does). Because we want to be alerted if this fails, we don't want to have to check.

Reply:
In 5666c6be82598066bdfe048239baf3fe2576c369 I updated the build pipeline so we only build the container image and publish to GHCR on merges to master/release/hotfix branches

Reply:
Yes I saw, thanks a lot. I've validated after reviewing